test/hotspot/jtreg/compiler/vectorization/runner/VectorizationTestRunner.java

```diff
@@ -54,31 +54,38 @@ public class VectorizationTestRunner {
     private static final long COMP_THRES_SECONDS = 30;
 
     protected void run() {
+        Class klass = getClass();
+
         // 1) Vectorization correctness test
-        // For each method annotated with @Test in the test method, this test runner
+        // For each method annotated with "@Test" in test classes, this test runner
         // invokes it twice - first time in the interpreter and second time compiled
         // by C2. Then this runner compares the two return values. Hence we require
         // each test method returning a primitive value or an array of primitive type.
-        // And each test method should not throw any exceptions.
-        Class klass = getClass();
-        for (Method method : klass.getDeclaredMethods()) {
-            try {
-                if (method.isAnnotationPresent(Test.class)) {
-                    verifyTestMethod(method);
-                    runTestOnMethod(method);
+        // Some VM options like "-Xcomp" may mess with the compiler control for the
+        // correctness check. We disable the check in these cases.
+        boolean use_intp = WB.getBooleanVMFlag("UseInterpreter");
+        boolean use_comp = WB.getBooleanVMFlag("UseCompiler");
+        boolean bg_comp = WB.getBooleanVMFlag("BackgroundCompilation");
+        if (use_intp && use_comp && bg_comp) {
```

Comment:
For `-Xbatch` we need check `BackgroundCompilation`. As we lock the compilation before running the test method in the interpreter to get reference result, the VM will hang if both background compilation is disabled and the compilation is locked by WhiteBox.

I just experimented an alternative approach that running reference result with C1 execution instead of interpretation. In this way we can run with background compilation off. What do you think of this new approach?